File: examples/vulnerable_test.rb
# Ruby test file with intentional vulnerabilities

require 'digest'

class VulnerableApp
  # CWE-78: Command Injection
  def execute_command(user_input)
    system("ls -la #{user_input}")
  end
  
  # CWE-89: SQL Injection
  def query_user(username)
    query = "SELECT * FROM users WHERE name = '#{params[:username]}'"
    User.where(query)
  end
  
  # CWE-79: XSS
  def display_greeting(name)
    render inline: "<h1>Hello, #{name}</h1>".html_safe
  end
  
  # CWE-327: Weak Crypto
  def hash_password(password)
    Digest::MD5.hexdigest(password)
  end
  
  # CWE-798: Hardcoded Credentials
  def connect_database
    password = "SuperSecret123"
    api_key = "sk_live_1234567890"
    # Use credentials
  end
  
  # CWE-502: Unsafe Deserialization
  def deserialize_data(data)
    Marshal.load(data)
  end
  
  # CWE-1321: Mass Assignment
  def create_user
    User.create(params[:user])
  end
end

File: examples/vulnerable_test.php
<?php
// PHP test file with intentional vulnerabilities

// CWE-78: Command Injection
function executeCommand($userInput) {
    shell_exec("ls -la " . $userInput);
}

// CWE-89: SQL Injection
function queryUser($username) {
    $conn = mysqli_connect("localhost", "user", "pass", "db");
    $query = "SELECT * FROM users WHERE name = '" . $_GET['username'] . "'";
    mysqli_query($conn, $query);
}

// CWE-79: XSS
function displayGreeting() {
    echo "Hello, " . $_GET['name'];
}

// CWE-22: Path Traversal
function readFile() {
    $filename = $_GET['file'];
    $content = file_get_contents("/var/www/" . $filename);
    echo $content;
}

// CWE-327: Weak Crypto
function hashPassword($password) {
    return md5($password);
}

// CWE-798: Hardcoded Credentials
$password = "admin123";
$apiKey = "pk_live_1234567890abcdef";

// CWE-502: Unsafe Deserialization
function deserializeData() {
    $data = unserialize($_POST['data']);
    return $data;
}
?>


